### DISCUSSION TITLE
How can I shift (rotate) a ribbon on an image ?

### DISCUSSION THREAD

**Discussion by myicq** [ImageMagick/ImageMagick#6558] [Help] [answered]
### ImageMagick version

7.x

### Operating system, version and so on

Windows 10

### Description

I am experimenting with seamless images (tile). 
For a specific purpose, I need to "shift" a horizontal part of an image, so that data wraps around the edges.
The blue lines below is just to illustrate the concept of "ribbon". 

Before:
![Before](https://github.com/ImageMagick/ImageMagick/assets/6152041/f3cb7b46-0627-42b9-89c3-128003bce831)

After: shift left or right by a certain amount, in px or pct.
![shiftafter](https://github.com/ImageMagick/ImageMagick/assets/6152041/588d4579-19c4-44eb-8d96-86376d8aac98)

**Comment by snibgo** [comment, marked as answer]
`-roll` does what you want. See https://imagemagick.org/script/command-line-options.php#roll . For example:
```
magick c:\web\im\123456.jpg -roll +100+0 x.png
```
You can use `fx` expressions, such as one-tenth of the width:
```
magick c:\web\im\123456.jpg -roll +%[fx:w*0.1]+0 x.png
```

This shifts the entire image, which works in your example. To shift just part of the image, you can crop the part, roll it, and composite it back to the input. For example:
```
magick 123456.jpg ( +clone -crop 244x49+0+194 -roll +100+0 ) -geometry +0+194 -composite x2.png
```

**Reply by GeeMack** [reply]
The "-roll" operator will also take arguments as percentages, like ` -roll +33.333+0%` .

**Comment by snibgo** [comment]
An alternative, slightly simpler:
```
magick c:\web\im\123456.jpg ( +clone -crop 244x49+0+194 -roll +100+0 ) -flatten x.png
```
